build.sh

```diff
@@ -28,6 +28,34 @@ json_path="project.json"
 os_names=$(jq -r '.os[] | .name' $json_path)
 os_array=($os_names)
 
+local_os=$(uname)
+
+check_dependencies() {
```

Comment:
Could this be simplified a little like in the download.sh?

Of course, this below would need to loop through the dependencies and verify the version is >= that in the JSON.

```sh
check_command() {
    local cmd=$1
    if command -v $cmd >/dev/null 2>&1; then
        local version=$($cmd --version | head -n 1)
        print_message info "$cmd is installed."
    else
        print_message error "$cmd is not installed."
        exit 1
    fi
}

# This function checks if the required dependencies are installed.
check_dependencies() {
    check_command curl
    check_command wget
    check_command jq
    printf "\nPress \033[32mEnter\033[0m to continue."
    read -r input
    if [[ -z "$input" ]]; then
        return
    else
        printf "\nPress \033[32mEnter\033[0m to continue."
    fi
}
```